pkg/sql/catalog/lease/lease.go

```diff
@@ -1655,14 +1668,24 @@ func (m *Manager) handleRangeFeedAvailability(ctx context.Context) {
 	// the range feed.
 	if timeutil.Since(m.mu.rangeFeedIsUnavailableAt) >=
 		LeaseMonitorRangeFeedResetTime.Get(&m.settings.SV) {
-		log.Warning(ctx, "attempting restart of leasing range feed")
-		// Attempt a range feed restart if it has been down too long.
-		m.watchForUpdates(ctx)
-		// Track when the last restart occurred.
-		m.mu.rangeFeedIsUnavailableAt = timeutil.Now()
+		m.restartLeasingRangeFeedLocked(ctx)
 	}
 }
 
+func (m *Manager) handleRangeFeedError(ctx context.Context) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.restartLeasingRangeFeedLocked(ctx)
+}
+
+func (m *Manager) restartLeasingRangeFeedLocked(ctx context.Context) {
+	log.Warning(ctx, "attempting restart of leasing range feed")
+	// Attempt a range feed restart if it has been down too long.
+	m.watchForUpdates(ctx)
+	// Track when the last restart occurred.
+	m.mu.rangeFeedIsUnavailableAt = timeutil.Now()
```

Comment:
Seems to me like we we should also be resetting the number of checkpoints but the existing code didn't.